### ISSUE TITLE
The new API should have a find() method for maps.

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jbeder/yaml-cpp#181]
```
This is a feature request.

The new API should have a few STL-like helper functions, like find().
```

Original issue reported on code.google.com by `jbe...@gmail.com` on 3 Jan 2013 at 8:17

**Comment by ebkalderon** [comment]
Definitely! I can't figure out an alternative for the older Node::FindValue() method.

**Comment by jbeder** [comment]
FYI, you can always just ask for the value via `node["foo"]` - it'll return convertible-to-false if it's not there.

The idea of this feature request was to return an iterator.

**Comment by till-s** [comment]
This is old but I'd like to second this request.

I know you can use operator[] but the overloads are **very** confusing:

node["foo"] actually inserts a new node into the map (and obsoletes iterators, as of 0.6.2)

whereas a pure lookup has to be coded

static_cast<const node>(node)["foo"];

It is very hard to identify places where a pure lookup was intended. A 'find' method would be much better than this quirky overload.
